### ISSUE TITLE
PTZ-Director (NDI PTZ camera controls app for macOS)

### ISSUE THREAD

**Issue by Ninakadin** [bitfocus/companion-module-requests#748]
Today PTZ-Director for macOS was released, an app to control up to 4 NDI PTZ cameras from a Mac. The app has USB-HID joystick support and having Bitfocus Companion in mind, LiveSkills has implemented a rich AppleScript Interface to provide a useful integration into Companion for Stream Decks.
Here is the documentation of the AppleScript commands: https://apps.apple.com/de/app/id1613397327
Each function is designed to be used as shortcut.
An evaluation copy can be downloaded via TestFlight (requires macOS 12.x): https://testflight.apple.com/join/7vdMSs7V

LiveSkills is happy to support anyone working on this. If you need support, contact: info@liveskills.net

**Comment by josephdadams** [comment]
Applescript is great, but requires a middleman app to be run currently. Be a lot more efficient if you could implement a TCP API of some kind to allow for direct control over the network.

**Comment by Ninakadin** [comment]
Am I right that you would prefer e.g. a TCP Telnet interface with a kind of "AT command“ like interface, so there is no need for Companion to have software running on that Mac? If this is the case, do you have common protocols that are already in use by Companion?  I mean, we don’t need to invent the wheel again ;-)
Using AppleScript of course would require an app to run on the Mac, on the other hand AppleScript is a standardised interface.
In general, yes, we could do that (with some minimal support, so that we take the right approach at first try), but would take some time.

**Comment by josephdadams** [comment]
Yes, something like that would be nice. If you can provide return data in the protocol, even better. Take a look at some of the Roland modules for a simple TCP protocol.

**Comment by Ninakadin** [comment]
@josephdadams Do you have a link to some documentation? Just the basics.

**Comment by josephdadams** [comment]
https://static.roland.com/assets/media/zip/XS-82H_83H_84H_reference_v30_eng03_W.pdf
https://github.com/bitfocus/companion-module-roland-xs84h/blob/master/src/index.js
